src/hooks/index.js

```diff
@@ -348,3 +348,93 @@ exports.checkSchoolOwnership = hook => {
 				throw new errors.Forbidden('You do not have valid permissions to access this.');
 		});
 };
+
+//TODO: later: Template building
+//z.B.: maildata.template = { path: "../views/template/mail_new-problem.hbs", "data": { "firstName": "Hannes", .... } };
+//if (maildata.template) { [Template-Build (view client/controller/administration.js)] }
+// mail.html = generatedHtml || "";
+exports.sendEmail = (hook, maildata) => {
+	const userService = hook.app.service('/users');
+	const mailService = hook.app.service('/mails');
+	
+	let roles = (typeof maildata.roles === "string" ? [maildata.roles] : maildata.roles) || [];
+	let emails = (typeof maildata.emails === "string" ? [maildata.emails] : maildata.emails) || [];
+	let userIDs = (typeof maildata.userIds === "string" ? [maildata.userIds] : maildata.userIds) || [];
+	let receipients = [];
+	
+	let promises = [];
+
+	if (roles.length > 0) {
+		promises.push(
+			userService.find({query: {
+				roles: [roles],
+				schoolId: hook.data.schoolId,
+				$populate: ['roles']
+			}})
+		);
+	}
+	
+	if (userIDs.length > 0){
+		userIDs.map (id => {
+			promises.push(
+				userService.find({query: {
+					_id: mongoose.Types.ObjectId(id)
+				}})
+			);
+		});
+	}
+	
+	if (emails.length > 0){
+		emails.map(email => {
+			let re = /\S+@\S+\.\S+/;
+    		if (re.test(email)){
+				receipients.push(email);
+			}
+		});
+	}
+
+	if(promises.length > 0){
+		Promise.all(promises)
+		.catch(err => {
+			let error = err;
```

Comment:
just a consistency thing again.
Promise.all(promises)
.then()
.catch();
Meaning you just have to switch it up.

You could also continue dealing with the error.
Meaning you either `return Promise.reject(new Error....)` or include `const errors = require('feathers-errors');` at the very top of the file and use an error from that like BadRequest with the message 'Problem resolving user or role'. That way the Process using your function knows something went wrong. Either Promise.reject or throw new error, either way should work.